### PR TITLE
Feature/UserInput submission UI

### LINKED ISSUE #108: Provide UI for UserInput submission
## Acceptance criteria:
 1. UI is implemented according to the [[﻿mockups ](https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=2154-8640&t=cTLlOUnzPEwMgJ4Z-0)](https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=2154-8640&t=cTLlOUnzPEwMgJ4Z-0)﻿
 2. UI components are functional
 3. UserInput is captured and sent to the backend

### PR DESCRIPTION
**Pull request by lomeintheory**
## Description

I decided to submit this PR as a draft since it might benefit from merging after ticket #86 is merged with dev and the new UserInput routes are in place. The only feature missing for this ticket is functionality that sends the user's input to the backend via the new routes.

Fixes #108 

## Type of change

- [x] ✨  New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Ran it on localhost to compare with our figma design

## Checklist:

- [x] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [ ] I have ensured that my pull request title is descriptive.

## Additional Information:

Currently using localStorage to store the checkbox's "checked" state in the browser. This is a temporary solution since I imagine we would want to store that state in a user's session after we have auth set up.

### CODE REVIEW

**Comment by relandboyle** [comment]
@lomeintheory Will you please link this PR to the corresponding issue and make sure the issue is assigned to you?

In the PR description, where you see "Fixes #..." you can delete the "(issue)" and start typing the issue number. You should see a list of matching issues pop up and you can select the right one.

**Review by relandboyle — APPROVED** [review]

File: client/components/layout/MainLayout.tsx [hunk 1]
```diff
@@ -211,15 +213,12 @@ const MainLayout: React.FC = () => {
           <Box></Box>
         )}
 
-        <Box
-          className="router-outlet"
-          sx={{ display: 'flex' }}
-        >
+        <Box className="router-outlet">
           <Typography
```
Comment (relandboyle) [inline client/components/layout/MainLayout.tsx:217]: @lomeintheory I think you want to remove the Typography element here and style your text within the PromptInputForm component.

File: client/components/common/PromptInputForm.tsx [hunk 2]
```diff
@@ -1,7 +1,136 @@
...
+    const savedCheckedState = localStorage.getItem('isChecked');
+    if (savedCheckedState === 'true') {
+      setIsChecked(true);
+    }
+  }, []);
+
+  const newUserInputMutation = useMutation({
+    mutationFn: async (input: UserInputPayload) => {
+      const response = await axios.post(
+        'http://localhost:3000/user-input/v1/inputs',
```
Comment (kinskiwu) [inline client/components/common/PromptInputForm.tsx:60]: Pls change it to `http://localhost:3000/v1/inputs`.

**Review by kinskiwu — COMMENTED** [review]
Thanks for the good work! Please review the comments and also remove the unnecessary placeholders from `PromptInputPage`.

**Review by kinskiwu — APPROVED** [review]
Good job!

**Comment by relandboyle** [comment]
@lomeintheory I know you've already resolved a bunch of comments on this PR but I think there is more to do.
@ChTiSh 

image

image

Here we have three interactive elements; text input, checkbox input, and button.
For each element we should account for active, focused, and hover states. For checkbox input we should account for checked and unchecked styling. For text input we should account for error state / input active and error state / input inactive. We should also style the cursor. For button, disabled style should probably affect background color as well as / instead of foreground color, and hover should not make it disappear.

As all of these are intended to be modular components used throughout the site, I think it would be worth the effort to define all of their states and styles in Figma.

Let me know how we should resolve this. Throw it all in one new issue for a future sprint? Multiple issues? Keep in mind, just because someone built it up this far does not mean the same person needs to make the additional changes.

**Comment by ChTiSh** [comment]
I would vote for issuing new different issues for modular reusable components. I will work on updating the figma and issuing the tickets. It might make sense to issue the tickets in this sprint as the input edit/update could use it.

**Comment by lomeintheory** [comment]
> @lomeintheory I know you've already resolved a bunch of comments on this PR but I think there is more to do. @ChTiSh
> 
> image
> 
> image
> 
> Here we have three interactive elements; text input, checkbox input, and button. For each element we should account for active, focused, and hover states. For checkbox input we should account for checked and unchecked styling. For text input we should account for error state / input active and error state / input inactive. We should also style the cursor. For button, disabled style should probably affect background color as well as / instead of foreground color, and hover should not make it disappear.
> 
> As all of these are intended to be modular components used throughout the site, I think it would be worth the effort to define all of their states and styles in Figma.
> 
> Let me know how we should resolve this. Throw it all in one new issue for a future sprint? Multiple issues? Keep in mind, just because someone built it up this far does not mean the same person needs to make the additional changes.

I think we talked about getting this PR merged to get a basic working input bar on our dev branch, so I definitely have no problem working more on the styling. @ChTiSh I'll make a new ticket for this and assign it to myself. End of the week so I have more time for Linkta.

**Comment by lomeintheory** [comment]
> kbox input, and button.

I was going to create a ticket for this but looks like you already 😅

**Comment by ChTiSh** [comment]
Haha, I created some drafts as placeholders, feel free to convert them to issues, edit/add anything.

**Pull request merged**